pkg/proctree/process_context.go

```diff
@@ -29,30 +28,26 @@ type ProcessCtx struct {
 }
 
 type ProcessTree struct {
-	processTreeMap map[int]ProcessCtx
+	ProcessTreeMap map[int]ProcessCtx
 }
```

Comment:
Let's not make this struct public (use uncapitalized struct and field names).
We should use getters/setters to access and update process info.
You can even take the changes introduced in #1452 into this PR and to make these getters/setters thread safe